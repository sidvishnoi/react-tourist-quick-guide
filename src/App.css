* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  background: gold;
}

.Content {
  padding: 1em;
}

h2.source {
  padding: 1em 1em 0;
  text-align: center;
}
