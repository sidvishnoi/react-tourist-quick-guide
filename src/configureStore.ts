import { createStore, applyMiddleware, compose, StoreEnhancer } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { devToolsEnhancer } from 'redux-devtools-extension';
import rootReducer from './reducers';

export default function configureStore(preloadedState?: any) {
  const middlewares = [thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const enhancers = [middlewareEnhancer, devToolsEnhancer({})];
  const composedEnhancers = compose(...enhancers) as StoreEnhancer<{}, {}>;

  const store = createStore(rootReducer, preloadedState, composedEnhancers);
  return store;
}
