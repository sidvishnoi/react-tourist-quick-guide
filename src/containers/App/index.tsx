import * as React from 'react';
import { connect } from 'react-redux';

import { addCity, getSuggestions, selectLocation } from '../../actions';
import { State } from '../../state';

import Text from '../../components/Text';
import AutoComplete from '../AutoComplete';
import Cities from '../Cities';

const mapStateToProps = (state: State) => ({
  source: state.source,
});

function App({ source }: { source: string }) {
  if (!source) {
    return (
      <div className="App">
        <Text as="h1" size="fsize_09">
          Where are you travelling from?
        </Text>
        <AutoComplete
          placeholder="Your Location"
          onChange={getSuggestions}
          onSelect={selectLocation}
        />
      </div>
    );
  }

  return (
    <div className="App">
      <Text as="h1" size="fsize_09">
        Destinations from <em>{source}</em>:
      </Text>
      <AutoComplete
        placeholder="Add destination"
        onChange={getSuggestions}
        onSelect={addCity}
      />
      <div className="Content">
        <Cities />
      </div>
    </div>
  );
}

export default connect(mapStateToProps)(App);
