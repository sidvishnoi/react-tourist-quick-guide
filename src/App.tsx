import * as React from 'react';
import { connect } from 'react-redux';

import { addCity, getSuggestions, selectLocation } from './actions';
import './App.css';
import AutoCompleteSearch from './containers/AutoCompleteSearch';
import Cities from './containers/Cities';
import { State } from './state';

const mapStateToProps = ({ source }: State) => ({
  source,
});

function App({ source }: { source: string }) {
  if (!source) {
    return (
      <div className="App">
        <div className="Search">
          <AutoCompleteSearch
            placeholder="Your Location"
            onChange={getSuggestions}
            onSelect={selectLocation}
          />
        </div>
      </div>
    );
  }

  return (
    <div className="App">
      <h2 className="source">
        Destinations from <em>{source}</em>:
      </h2>
      <div className="Search">
        <AutoCompleteSearch
          placeholder="Add destination"
          onChange={getSuggestions}
          onSelect={addCity}
        />
      </div>
      <div className="Content">
        <Cities />
      </div>
    </div>
  );
}

export default connect(mapStateToProps)(App);
