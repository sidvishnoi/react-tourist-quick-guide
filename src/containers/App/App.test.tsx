import { mount } from 'enzyme';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import { Provider } from 'react-redux';
import * as renderer from 'react-test-renderer';

import App from '.';
import configureStore from '../../configureStore';
import initialState from '../../state';

Enzyme.configure({ adapter: new Adapter() });

describe('Containers - App', () => {
  it('renders initial page when source is not available', () => {
    const store = configureStore(initialState);
    const app = (
      <Provider store={store}>
        <App />
      </Provider>
    );

    const output = mount(app);
    expect(output.find('h1').text()).toEqual('Where are you travelling from?');

    const tree = renderer.create(app);
    expect(tree).toMatchSnapshot();
  });

  it('renders destination search page when source is set', () => {
    const store = configureStore({ ...initialState, source: 'PASS' });
    const app = (
      <Provider store={store}>
        <App />
      </Provider>
    );

    const output = mount(app);
    expect(output.find('h1').text()).toEqual('Destinations from PASS:');

    const tree = renderer.create(app);
    expect(tree).toMatchSnapshot();
  });
});
