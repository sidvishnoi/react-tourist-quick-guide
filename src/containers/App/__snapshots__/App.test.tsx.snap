// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers - App renders destination search page when source is set 1`] = `
<div
  className="App"
>
  <h1
    className="sc-bdVaJa fCEYOb"
    color="#000"
    size="fsize_09"
  >
    Destinations from 
    <em>
      PASS
    </em>
    :
  </h1>
  <div
    className="sc-bwzfXH fQaCWc"
  >
    <div
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <input
        aria-autocomplete="list"
        aria-expanded={false}
        autoComplete="off"
        className=""
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add destination"
        role="combobox"
        value=""
      />
    </div>
  </div>
  <div
    className="Content"
  >
    <div
      className="sc-ifAKCX ulTBn"
    >
      Advertisement Space
    </div>
  </div>
</div>
`;

exports[`Containers - App renders initial page when source is not available 1`] = `
<div
  className="App"
>
  <h1
    className="sc-bdVaJa fCEYOb"
    color="#000"
    size="fsize_09"
  >
    Where are you travelling from?
  </h1>
  <div
    className="sc-bwzfXH fQaCWc"
  >
    <div
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <input
        aria-autocomplete="list"
        aria-expanded={false}
        autoComplete="off"
        className=""
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Your Location"
        role="combobox"
        value=""
      />
    </div>
  </div>
</div>
`;
